pkg/istiodcert/worker.go

```diff
@@ -57,7 +57,7 @@ type DynamicIstiodCertProvisioner struct {
 
 	issuerRefMutex sync.Mutex
 
-	issuerChangeChan <-chan *cmmeta.ObjectReference
+	issuerChangeChan *certmanager.IssuerChangeSubscription
```

Comment:
nitpick (non-blocking): maybe rename because this isn't a channel any more?

```suggestion
	issuerChangeSub *certmanager.IssuerChangeSubscription
```